scripts/paypal/mark-double-transactions-as-refunded.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ * This script can be used whenever PayPal webhooks event types change to update
+ * Host's connected accounts.
+ */
+
+import '../../server/env';
+
+import logger from '../../server/lib/logger';
+import { createRefundTransaction } from '../../server/lib/payments';
+import models, { sequelize } from '../../server/models';
+
+/**
+ * Refund the wrongly recorded double transactions for PayPal payment, to make sure
+ * they will be deducted from the next settlement.
+ */
+const main = async (): Promise<void> => {
+  const doubleTransactionsInfos = await sequelize.query(
+    `
+    SELECT t."OrderId", array_agg(DISTINCT t.id) AS "transactionIds"
+    FROM "Transactions" t 
+    INNER JOIN "PaymentMethods" pm ON t."PaymentMethodId" = pm.id
+    INNER JOIN "Orders" o ON t."OrderId" = o.id
+    WHERE pm.service = 'paypal'
+    AND t.kind = 'CONTRIBUTION'
+    AND t."type" = 'CREDIT'
+    AND t."deletedAt" IS NULL
+    AND t."isRefund" IS FALSE
+    AND t."RefundTransactionId" IS NULL
+    GROUP BY t."OrderId", date_trunc('day', t."createdAt") 
+    HAVING count(t.id) > 1
+    ORDER BY date_trunc('day', t."createdAt") DESC
+  `,
+    { raw: true, type: sequelize.QueryTypes.SELECT },
+  );
+
+  for (const doubleTransactionInfo of doubleTransactionsInfos) {
+    const sortedTransactionIds = [...doubleTransactionInfo.transactionIds].sort();
+    const transactionIdsToRefund = sortedTransactionIds.slice(1); // Keep the first transaction
```

Comment:
In case you want to be sure it's the first transaction, you need to order by Transaction.id ASC.

Reply:
Oh, it's already sorted, sorry.